csharp/src/Google.Protobuf/Reflection/ReflectionUtil.cs

```diff
@@ -304,7 +328,28 @@ public void ClearExtension(IMessage message)
             }
         }
 
-        private class ExtensionSetReflector<
+#if NET6_0_OR_GREATER
+        private sealed class AotReflectionHelper : IReflectionHelper
+        {
+            private static readonly object[] EmptyObjectArray = new object[0];
+            public Action<IMessage> CreateActionIMessage(MethodInfo method) => message => method.Invoke(message, EmptyObjectArray);
+            public Action<IMessage, object> CreateActionIMessageObject(MethodInfo method) => (message, arg) => method.Invoke(message, new object[] { arg });
+            public Func<IMessage, bool> CreateFuncIMessageBool(MethodInfo method) => message => (bool) method.Invoke(message, EmptyObjectArray);
+            public Func<IMessage, int> CreateFuncIMessageInt32(MethodInfo method) => message => (int) method.Invoke(message, EmptyObjectArray);
+            public Func<IMessage, object> CreateFuncIMessageObject(MethodInfo method) => message => method.Invoke(message, EmptyObjectArray);
+        }
+
+        private sealed class AotExtensionReflectionHelper : IExtensionReflectionHelper
```

Comment:
Please could you add a comment explaining *why* it's not supported? (That way, if things change we could potentially fix it later.)